src/Compilers/CSharp/Portable/Binder/Semantics/AccessCheck.cs

```diff
@@ -57,6 +56,56 @@ internal static class AccessCheck
             return IsSymbolAccessibleCore(symbol, within, throughTypeOpt, out failedThroughTypeCheck, within.DeclaringCompilation, ref useSiteDiagnostics, basesBeingResolved);
         }
 
+        /// <summary>
+        /// Returns true if the symbol is effectively public or internal based on
+        /// the declared accessibility of the symbol and any containing symbols.
+        /// </summary>
+        internal static bool IsPublicOrInternal(Symbol symbol, out bool isInternal)
```

Comment:
>IsPublicOrInternal [](start = 29, length = 18)

Consider including "effectively" in the name for clarity. Something like "Is**Effectively**PublicOrInternal".